src/ol/interaction/extentinteraction.js

```diff
@@ -0,0 +1,429 @@
+        if (this.snappedToVertex_) {
+          vertex = squaredDist1 > squaredDist2 ?
+              closestSegment[1] : closestSegment[0];
+        }
+        return vertex;
+      }
+    }
+    return null;
+  };
+  this.handlePointerMove = function(mapBrowserEvent) {
+    var pixel = mapBrowserEvent.pixel;
+    var map = mapBrowserEvent.map;
+
+    var vertex = this.snapToVertex(pixel, map);
+    if (!vertex) {
+      vertex = map.getCoordinateFromPixel(pixel);
+    }
+    this.createOrUpdatePointerFeature(vertex);
+  };
+  this.createOrUpdateExtentFeature = function(extent) {
+    var bboxFeature = this.bboxFeature_;
+
+    if (!extent) {
+      extent = [0,0,0,0];
+    }
+    if (!bboxFeature) {
+      bboxFeature = new ol.Feature(new ol.geom.Polygon([[
+        [extent[0], extent[1]],
+        [extent[2], extent[1]],
+        [extent[2], extent[3]],
+        [extent[0], extent[3]],
+        [extent[0], extent[1]]
+      ]], ol.geom.GeometryLayout.XY));
+
+      this.bboxFeature_ = bboxFeature;
+      this.bboxOverlay_.getSource().addFeature(bboxFeature);
+    } else {
+      var geometry = (bboxFeature.getGeometry());
+      geometry.setCoordinates([[
+        [extent[0], extent[1]],
+        [extent[2], extent[1]],
+        [extent[2], extent[3]],
+        [extent[0], extent[3]],
+        [extent[0], extent[1]]
+      ]]);
+    }
+    return bboxFeature;
+  };
+  this.removeExtentFeature = function() {
+    var bboxFeature = this.bboxFeature_;
+    if (bboxFeature) {
+      this.bboxOverlay_.getSource().removeFeature(bboxFeature);
+      this.bboxFeature_ = null;
+    }
+  };
+  this.createOrUpdatePointerFeature = function(vertex) {
+    var vertexFeature = this.vertexFeature_;
+    if (!vertexFeature) {
+      vertexFeature = new ol.Feature(new ol.geom.Point(vertex));
+      this.vertexFeature_ = vertexFeature;
+      this.vertexOverlay_.getSource().addFeature(vertexFeature);
+    } else {
+      var geometry = (vertexFeature.getGeometry());
+      geometry.setCoordinates(vertex);
+    }
+    return vertexFeature;
+  };
+  this.removePointerFeature = function() {
+    var vertexFeature = this.vertexFeature_;
+    if (vertexFeature) {
+      this.vertexOverlay_.getSource().removeFeature(vertexFeature);
+      this.vertexFeature_ = null;
+    }
+  };
+};
+goog.inherits(ol.interaction.Extent, ol.interaction.Pointer);
+
+/* Public Methods */
+/**
+ * @inheritDoc
+ */
+ol.interaction.Extent.prototype.setMap = function(map) {
+  this.bboxOverlay_.setMap(map);
+  this.vertexOverlay_.setMap(map);
+  ol.interaction.Interaction.prototype.setMap.call(this, map);
+};
+/**
+ * Returns the current drawn extent in the requested projection
+ *
+ * @param {ol.proj.Projection=} projection Projection to retun the extent in
+ * @return {ol.Extent} Drawn extent in the passed projection, or in the native projection if undefined.
+ */
+ol.interaction.Extent.prototype.getExtent = function(projection) {
```

Comment:
setExtent is also intended to be public.

Sample use case is if you have an extent that you can set by text field and this interaction, you want each to be able to update the other.